src/metrics.go

```diff
@@ -200,7 +209,8 @@ func getMetricsData(sample *metric.Set) error {
 		}
 		return populateMetrics(sample, rawMetrics, metricsDefinition)
 	case httpAPIStatus:
-		for _, p := range strings.Split(args.Endpoints, ",") {
```

Comment:
In the config sample, the config option Endpoints is defined as
          # endpoints: /nginx,/processes,/connections,/ssl,/slabs,/http,/http/requests,/http/server_zones,/http/caches,/http/upstreams,/http/keyvals,/stream,/stream/server_zones,/stream/upstreams,/stream/keyvals,/stream/zone_sync

and now the list seems limited to 

paths := []string{"/nginx", "/processes", "/connections", "/http/requests", "/ssl"}

This looks like a big breaking change for existing nginx plus customers. How can they collect all the other endpoints?

Reply:
Yes, it's true. The problem is that if we keep them and users enable them, not only they will get bad data, they may even lose "good" data because of the 240 attribute limit on the samples, which I think get removed "randomly" above 240.
By default I think endpoints they are not enabled and not even documented. We can increase the version and notify that these endpoints were deprecated, until we have the final solution.
We will also document officially that they are not supported.

Other option is to leave them as is and deal with the consequences if they come up. What do you think?

Reply:
well, maybe we could update the config and reduce the number of endpoints by default. Or we could split the metrics in two different configurations (for instance, one of them queries /nginx,/processes,/connections,/ssl,/slabs,/http,/http/requests,/http/server_zones,/http/caches, and the other one /http/upstreams,/http/keyvals,/stream,/stream/server_zones,/stream/upstreams,/stream/keyvals,/stream/zone_sync)

I wouldn't remove the Endpoints config option though. Maybe we should discuss this with the team.